barretenberg/cpp/src/barretenberg/stdlib/primitives/group/cycle_group.cpp

```diff
@@ -0,0 +1,1322 @@
+#include "../field/field.hpp"
+#include "barretenberg/crypto/pedersen_commitment/pedersen.hpp"
+#include "barretenberg/ecc/curves/grumpkin/grumpkin.hpp"
+
+#include "../../hash/pedersen/pedersen.hpp"
+#include "../../hash/pedersen/pedersen_gates.hpp"
+
+#include "./cycle_group.hpp"
+#include "barretenberg/proof_system/plookup_tables/types.hpp"
+#include "barretenberg/stdlib/primitives/plookup/plookup.hpp"
+namespace proof_system::plonk::stdlib {
+
+template <typename Composer>
+cycle_group<Composer>::cycle_group(Composer* _context)
+    : context(_context)
+    , x(0)
+    , y(0)
+    , _is_infinity(true)
+    , _is_constant(true)
+{}
+
+/**
+ * @brief Construct a new cycle group<Composer>::cycle group object
+ *
+ * @param _x
+ * @param _y
+ * @param is_infinity
+ */
+template <typename Composer>
+cycle_group<Composer>::cycle_group(field_t _x, field_t _y, bool_t is_infinity)
+    : context(_x.get_context() == nullptr
+                  ? _y.get_context() == nullptr
+                        ? is_infinity.get_context() == nullptr ? nullptr : is_infinity.get_context()
+                        : _y.get_context()
+                  : _x.get_context())
+    , x(_x.normalize())
+    , y(_y.normalize())
+    , _is_infinity(is_infinity)
+    , _is_constant(_x.is_constant() && _y.is_constant() && is_infinity.is_constant())
+{}
```

Comment:
```suggestion
template <typename Composer>
cycle_group<Composer>::cycle_group(field_t _x, field_t _y, bool_t is_infinity)
    : x(_x.normalize())
    , y(_y.normalize())
    , _is_infinity(is_infinity)
    , _is_constant(_x.is_constant() && _y.is_constant() && is_infinity.is_constant())
{
    if (_x.get_context() != nullptr) {
        context = _x.get_context();
    } else if (_y.get_context() != nullptr) {
        context = _y.get_context();
    } else {
        context = is_infinity.get_context();
    }
}
```

Reply:
If its possible to do something like this above, its much easier to read versus the nested ternary operators

Reply:
Yup, applied. Needed to shuffle the ordering of member variables but should be fine